### PR TITLE
Update Testing Documentation

### PR DESCRIPTION
**Pull request by Jdubrick**
**Please specify the area for this PR**
Documentation updates

**What does does this PR do / why we need it**:
This PR updates the documentation for how to properly test the Devfile Registry Operator. This also updates the PR template to include these instructions so reviewers will always have that information readily available.

**Which issue(s) this PR fixes**:

No issue

**PR acceptance criteria**:

- [ ] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [ ] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [ ] Does the registry operator documentation need to be updated with your changes?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Review by michael-valdron — COMMENTED** [review]
Changes looks good.

We should include a note for the integration test instructions stating this is subject to change when https://github.com/devfile/api/issues/1523 is resolved.

**Review by michael-valdron — APPROVED** [review]
/lgtm

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -16,4 +16,23 @@ Fixes #?
...
+-->
+**Testing changes**
+
+Unit Tests:
+- Run `make test` from root of the repository.
+
+<!--
+Will need to be updated after the completion of https://github.com/devfile/api/issues/1523
+-->
+Integration Tests:
```
Comment (thepetk) [inline .github/PULL_REQUEST_TEMPLATE.md:30]: looks good to me and very useful info I have to say.

I was wondering if it makes sense to:
* Make the integration tests instructions a section in contributing.md
* Replace the text here with a link to contributing.md

This way we only have to update contributing guide if we want to make any change. WDYT?

**Review by thepetk — COMMENTED** [review]
Left a minor comment

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (michael-valdron) [reply .github/PULL_REQUEST_TEMPLATE.md:30]: +1 on @thepetk suggestion, @Jdubrick we can the link to the Integration Tests section under `CONTRIBUTING.md` which also contains your changes, similar to what you did in the README.

Comment (Jdubrick) [reply .github/PULL_REQUEST_TEMPLATE.md:30]: Good idea thanks for catching that!

**Review by thepetk — APPROVED** [review]
/lgtm

**Pull request merged**